### PR TITLE
Broker unityfork

### PR DESCRIPTION
**Pull request by RPangrle-MS**
Merge changes from the unity fork into broker
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/314%23issuecomment-101478849%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/314%23discussion_r30199543%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/314%23discussion_r30199606%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/314%23discussion_r30199833%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/314%23discussion_r30199916%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/314%23issuecomment-101478849%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Hi%20__%40RPangrle-MS__%2C%20I%27m%20your%20friendly%20neighborhood%20Microsoft%20Open%20Technologies%2C%20Inc.%20Pull%20Request%20Bot%20%28You%20can%20call%20me%20MSOTBOT%29.%20Thanks%20for%20your%20contribution%21%5Cr%5Cn%20%20%20%20%3Cspan%3EYou%27ve%20already%20signed%20the%20contribution%20license%20agreement.%20Thanks%21%3C/span%3E%5Cr%5Cn%20%20%20%20%20%20%20%20%3Cp%3EThe%20agreement%20was%20validated%20by%20Microsoft%20Open%20Technologies%2C%20Inc.%20and%20real%20humans%20are%20currently%20evaluating%20your%20PR.%3C/p%3E%5Cr%5Cn%5Cr%5CnTTYL%2C%20MSOTBOT%3B%5Cr%5Cn%22%2C%20%22created_at%22%3A%20%222015-05-13T01%3A32%3A07Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9519461%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/msotclas%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20d41f46dc14c52cb54ed89ced5c54385a8ad3b85a%20ADALiOS/ADALiOS/ADAuthenticationWebViewController.m%2042%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/314%23discussion_r30199543%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22sorry.%20my%20bad.%20this%20needs%20to%20die.%20remove%20this%20if%20block%20as%20this%20was%20moved%20to%20%5BADCustomHeaderHandler%5D%22%2C%20%22created_at%22%3A%20%222015-05-13T03%3A02%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/5713915%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kpanwar%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222015-05-13T03%3A12%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7004783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RPangrle-MS%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20ADALiOS/ADALiOS/ADAuthenticationWebViewController.m%3AL161-188%22%7D%2C%20%22Pull%20d41f46dc14c52cb54ed89ced5c54385a8ad3b85a%20ADALiOS/ADALiOS/ADAuthenticationWebViewController.m%2021%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/314%23discussion_r30199606%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22in%20line%20152%20add%20a%20comment%20that%20msauth%20comparison%20will%20not%20hit%22%2C%20%22created_at%22%3A%20%222015-05-13T03%3A05%3A08Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/5713915%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kpanwar%22%7D%7D%2C%20%7B%22body%22%3A%20%22Going%20to%20move%20this%20into%20the%20mentioned%20block%2C%20the%20resulting%20code%20will%20be%20clearer.%20%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222015-05-13T03%3A14%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7004783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RPangrle-MS%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20ADALiOS/ADALiOS/ADAuthenticationWebViewController.m%3AL161-188%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/314#issuecomment-101478849'>General Comment</a></b>
- <a href='https://github.com/msotclas'></a> Hi __@RPangrle-MS__, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
<span>You've already signed the contribution license agreement. Thanks!</span>
<p>The agreement was validated by Microsoft Open Technologies, Inc. and real humans are currently evaluating your PR.</p>
TTYL, MSOTBOT;
- [x] <a href='#crh-comment-Pull d41f46dc14c52cb54ed89ced5c54385a8ad3b85a ADALiOS/ADALiOS/ADAuthenticationWebViewController.m 42'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/314#discussion_r30199543'>File: ADALiOS/ADALiOS/ADAuthenticationWebViewController.m:L161-188</a></b>
- <a href='https://github.com/kpanwar'></a> sorry. my bad. this needs to die. remove this if block as this was moved to [ADCustomHeaderHandler]
- [x] <a href='#crh-comment-Pull d41f46dc14c52cb54ed89ced5c54385a8ad3b85a ADALiOS/ADALiOS/ADAuthenticationWebViewController.m 21'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/314#discussion_r30199606'>File: ADALiOS/ADALiOS/ADAuthenticationWebViewController.m:L161-188</a></b>
- <a href='https://github.com/kpanwar'></a> in line 152 add a comment that msauth comparison will not hit
- <a href='https://github.com/RPangrle-MS'></a> Going to move this into the mentioned block, the resulting code will be clearer. :+1:

<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-objc/pull/314?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-objc/pull/314?approve=1'></a>&nbsp;<a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/314'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by brandwe** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/314'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by msotclas** [comment]
Hi __@RPangrle-MS__, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft Open Technologies, Inc. and real humans are currently evaluating your PR.</p>

TTYL, MSOTBOT;

File: ADALiOS/ADALiOS/ADAuthenticationWebViewController.m [hunk 1]
```diff
@@ -153,12 +161,27 @@ - (BOOL)webView:(UIWebView *)webView shouldStartLoadWithRequest:(NSURLRequest *)
...
+        NSMutableURLRequest *mutableRequest = [request mutableCopy];
+        [mutableRequest addValue:pKeyAuthHeaderVersion forHTTPHeaderField:pKeyAuthHeader];
+        
+        // Now set our request variable with an (immutable) copy of the altered request
+        request = [mutableRequest copy];
+        [webView loadRequest:request];
+        return NO;
+    }
+#endif // AD_BROKER
+
```
Comment (kpanwar) [inline ADALiOS/ADALiOS/ADAuthenticationWebViewController.m]: sorry. my bad. this needs to die. remove this if block as this was moved to [ADCustomHeaderHandler]

File: ADALiOS/ADALiOS/ADAuthenticationWebViewController.m [hunk 2]
```diff
@@ -153,12 +161,27 @@ - (BOOL)webView:(UIWebView *)webView shouldStartLoadWithRequest:(NSURLRequest *)
         // This event is explicitly scheduled on the main thread as it is UI related.
         NSAssert( nil != _delegate, @"Delegate object was lost" );
         
-        dispatch_async( dispatch_get_main_queue(), ^{ [_delegate webAuthenticationDidCompleteWithURL:request.URL]; } );
```
Comment (kpanwar) [inline ADALiOS/ADALiOS/ADAuthenticationWebViewController.m]: in line 152 add a comment that msauth comparison will not hit

File: ADALiOS/ADALiOS/ADAuthenticationWebViewController.m (hunk 1 above)
Comment (RPangrle-MS) [inline ADALiOS/ADALiOS/ADAuthenticationWebViewController.m]: :+1:

File: ADALiOS/ADALiOS/ADAuthenticationWebViewController.m (hunk 2 above)
Comment (RPangrle-MS) [inline ADALiOS/ADALiOS/ADAuthenticationWebViewController.m]: Going to move this into the mentioned block, the resulting code will be clearer. :+1:

**Pull request merged**